src/towncrier/build.py

```diff
@@ -213,6 +214,16 @@ def __main(
     # in the template.
     render_title = config.title_format == ""
 
+    # Add format-specific context to the template
+    md_header_level: int | None = None
```

Comment:
Maybe start with a default value that will work with the default template

```suggestion

    # This is the default value that should match the desing
    # of the current default template for Markdown.
    md_header_level = 1
```